realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -515,6 +517,17 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
 
             updateAsyncQueriesTask = null;
         }
+
+        executeAsyncTransactionCallbacks();
```

Comment:
I would like to simplify the logic by just calling executeAsyncTransactionCallbacks in HandlerController.realmChanged() the else clause there.